_playwright-tests/UI/SnapshotRepo.spec.ts

```diff
@@ -0,0 +1,108 @@
+    });
+
+    await test.step('Select snapshotting', async () => {
+      await page.getByRole('radio', { name: 'Snapshotting' }).first().click();
+    });
+
+    await test.step('Fill in the repository details', async () => {
+      await page.getByLabel('Name').fill(repoName);
+      await page
+        .getByLabel('URL')
+        .fill('https://jlsherrill.fedorapeople.org/fake-repos/revision/' + repoName);
+    });
+
+    await test.step('Filter by architecture', async () => {
+      await page.getByRole('button', { name: 'filter architecture' }).click();
+      await page.getByRole('option', { name: 'x86_64' }).click();
+    });
+
+    await test.step('Filter by version', async () => {
+      const versionFilterButton = page.getByRole('button', { name: 'filter version' });
+      await versionFilterButton.click();
+      await page.getByRole('menuitem', { name: 'el9' }).locator('label').click();
+      await page.getByRole('menuitem', { name: 'el8' }).locator('label').click();
+      await versionFilterButton.click(); // Close the version filter dropdown
+    });
+
+    await test.step('Wait for the successful API call', async () => {
+      if (await page.locator('.pf-v5-c-helper-text__item.pf-m-error').isVisible()) {
+        throw new Error('Error message in element is visible');
+      }
+    });
+
+    await test.step('Submit the form and wait for modal to disappear', async () => {
+      await Promise.all([
+        // Click on 'Save'
+        page.getByRole('button', { name: 'Save' }).first().click(),
+        page.waitForResponse(
+          (resp) =>
+            resp.url().includes('/bulk_create/') && resp.status() >= 200 && resp.status() < 300,
+        ),
+        expect(page.locator('div[id^="pf-modal-part"]').first()).not.toBeVisible(),
+      ]);
+    });
+
+    await test.step('Verify that snapshot is successful', async () => {
+      await expect(page.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+
+    await test.step('Verify that snapshot is in snapshots list', async () => {
+      await page.getByRole('button', { name: 'Kebab toggle' }).first().click();
+      await page.getByRole('menuitem', { name: 'View all snapshots' }).click();
+      await expect(page.getByLabel('SnapshotsView list of').locator('tbody')).toBeVisible();
+      const snapshotTimestamp = await page
+        .getByLabel('SnapshotsView list of')
+        .locator('tbody')
+        .textContent();
+      if (snapshotTimestamp != null) {
+        if ((await validateSnapshotTimestamp(snapshotTimestamp, 10)) == false) {
+          throw new Error('Most recent snapshot timestamp is older than 10 minutes!');
+        }
+      } else {
+        throw new Error('Snapshot timestamp not found!');
+      }
+      await page.getByLabel('Close', { exact: true }).click();
+    });
+
+    await test.step('Delete created repository', async () => {
+      const row = await getRowByName(page, repoName);
```

Comment:
here it would be better to use an url for the repo filtering 🔧 
the `getRowByName()` can also accept URL and it is better to filter by that rather than the name (unless the name is an UUID) 😅 
- _because: if you have 2 repos with names 'one' and 'test one', it will return list both and thus the kebab click will throw, because it won't know which one to select_

Reply:
>  'one' and 'test one', it will return list both

Maybe we want to use `exact: true`? :thinking: